src/interval_sets.jl

```diff
@@ -1,28 +1,73 @@
 
 ###### Set-related Helpers #####
 """
-    IntervalSet{T<:AbstractInterval} <: AbstractSet{T}
-
-A set type for performing multi-interval specific operations.
-https://en.wikipedia.org/wiki/Interval_arithmetic#Interval_operators
+    IntervalSet{T<:AbstractInterval}
+
+An set of points represented by a sequence of intervals. Set operations over interval sets
+return a new IntervalSet, with the fewest number of intervals possible. Unbounded intervals
+are not supported. The individual intervals in the set can be accessed using the iteration
+API or by passing the set to `Array`.
+
+see also: https://en.wikipedia.org/wiki/Interval_arithmetic#Interval_operators
+
+## Examples
+
+```jldoctest
+julia> using Intervals
+julia> convert(Array, union(IntervalSet(1..5), IntervalSet(3..8)))
+1-element Vector{Interval{Int64, Closed, Closed}}:
+ Interval{Int64, Closed, Closed}(1, 8)
```

Comment:
For these docstrings why not just have it display the `IntervalSet`?

Reply:
Because currently IntervalSet doesn’t have nice printing. See comment above.